cf-agent/cf-agent.c

```diff
@@ -1981,6 +1981,8 @@ static int AutomaticBootstrap(GenericAgentConfig *config)
         if (strlen(ipaddr) < sizeof(POLICY_SERVER_IP))
         {
             config->agent_specific.agent.bootstrap_argument = xstrdup(ipaddr);
+            config->agent_specific.agent.bootstrap_ip       = xstrdup(ipaddr);
+            config->agent_specific.agent.bootstrap_host     = xstrdup(ipaddr);
```

Comment:
I don't know how to test the avahi stuff, but I populated the variables to be more similar to a regular bootstrap.